plugins/woocommerce/src/Internal/Admin/ProductForm/Form.php

```diff
@@ -0,0 +1,122 @@
+<?php
+/**
+ * WooCommerce Product Form
+ *
+ * @package Woocommerce ProductForm
+ */
+
+namespace Automattic\WooCommerce\Internal\Admin\ProductForm;
+
+/**
+ * Contains logic for the WooCommerce Product Form.
+ */
+class Form {
+
+	/**
+	 * Class instance.
+	 *
+	 * @var Form instance
+	 */
+	protected static $instance = null;
+
+	/**
+	 * Store form fields.
+	 *
+	 * @var array
+	 */
+	protected static $form_fields = array();
+
+	/**
+	 * Get class instance.
+	 */
+	final public static function instance() {
+		if ( ! static::$instance ) {
+			static::$instance = new static();
+		}
+		return static::$instance;
+	}
+
+	/**
+	 * Init.
+	 */
+	public function init() {    }
+
+	/**
+	 * Adds a child menu item to the navigation.
+	 *
+	 * @param string $id Field id.
+	 * @param string $plugin_id Plugin id.
+	 * @param array  $args Array containing the necessary arguments.
+	 *     $args = array(
+	 *       'type'            => (string) Field type. Required.
+	 *       'section'         => (string) Field location. Required.
+	 *       'order'           => (int) Field order.
+	 *       'properties'      => (array) Field properties.
+	 *     ).
+	 */
+	public static function add_field( $id, $plugin_id, $args ) {
```

Comment:
Should this return the newly created instance of `Field` or a `WP_Error` if arguments are missing?

It might be useful to consumers and would make testing slightly easier.

Reply:
Yeah that's a good point, I will update that.